### PR TITLE
Version2020041302

### PR DESCRIPTION
**Pull request by MaksimBurnin**

### CODE REVIEW

File: classes/common.php [hunk 1]
```diff
@@ -4,16 +4,29 @@
...
 class common {
-    public static function reset_entry($conditions){
+    public static function reset_entry($conditions, $force = false){
         global $DB;
+
         $oldentry = $DB->get_record('availability_examus', $conditions);
-        if ($oldentry and $oldentry->status != 'Not inited') {
+
+        $not_inited = $oldentry && $oldentry->status == 'Not inited';
+
```
Comment (shofel) [inline classes/common.php:13]: Mixing `&&` and `||` in one condition is quiet hard to understand.
I suggest to split and reverse the condition:
```suggestion
if (!$oldentry) return false
else if ($not_inited && !$force) return false
else {
```

File: classes/common.php [hunk 2]
```diff
@@ -4,16 +4,29 @@
...
+            ]);
+
+            if (count($entries) == 0 || $force) {
+                if($force){
+                    foreach($entries as $old){
+                        $old->status = "Force reset";
+                        $DB->update_record('availability_examus', $old);
+                    }
+                }
+
```
Comment (shofel) [inline classes/common.php:29]: If count of `$entries` is zero, then `foreach` on `$entries` do nothing. It means, we could safely move this code outside of the `if` clause.

```suggestion
           if($force){
                foreach($entries as $old){
                    $old->status = "Force reset";
                    $DB->update_record('availability_examus', $old);
                }
            }
                
            if (count($entries) == 0 || $force) {


```

File: classes/common.php [hunk 3]
```diff
@@ -4,16 +4,29 @@
 defined('MOODLE_INTERNAL') || die();
 
 class common {
-    public static function reset_entry($conditions){
+    public static function reset_entry($conditions, $force = false){
         global $DB;
+
         $oldentry = $DB->get_record('availability_examus', $conditions);
```
Comment (shofel) [inline classes/common.php:10]: We get one record, but there could be more than one. Should we get them all?

File: classes/condition.php [hunk 4]
```diff
@@ -58,15 +59,19 @@ class condition extends \core_availability\condition {
     public function __construct($structure) {
         $manual_modes = ['normal', 'identification'];
 
-        if (!empty($structure->duration)) {
+        if(!empty($structure->duration)) {
```
Comment (shofel) [inline classes/condition.php:62]: According to the Moodle code style, spacing was already correct.
https://docs.moodle.org/dev/Coding_style#If_.2F_else

**Review by shofel — CHANGES_REQUESTED** [review]
I haven't reviewed everything yet.

I was trying to simplify the `if-else` logic at `reset_entry`, but finally came up with [full rewrite](https://github.com/examus/moodle-availability_examus/pull/17/files) of the function.

Shame on me, I didn't run the code, and even didn't check it with a linter.
Please, approve if the logic is preserved the same, and feel free accepting that PR into this branch (I mean `Version2020041302`)

File: classes/condition.php (hunk 4 above)
Comment (MaksimBurnin) [reply classes/condition.php:62]: Fixed in 831a346

**Pull request merged**